architecture.md

```diff
@@ -1,11 +1,86 @@
+- [Example: MongoDB Version Upgrade](#example-mongodb-version-upgrade)
+- [MongoDB Docker Images](#mongodb-docker-images)
 
+## Cluster Configuration
+
+You create and update MongoDB resources by defining a MongoDB resource definition. When you apply the MongoDB resource definition to your Kubernetes environment, the Operator:
+
+1. Creates a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) that contains one [pod](https://kubernetes.io/docs/concepts/workloads/pods/pod-overview/) for each [replica set](https://docs.mongodb.com/manual/replication/) member. 
+1. Creates two [containers](https://kubernetes.io/docs/concepts/containers/overview/) in each pod:
+
+   - A container for the [`mongod`](https://docs.mongodb.com/manual/reference/program/mongod/index.html) process binary. </br>
+     `mongod` is the primary daemon process for the MongoDB system. It handles data requests, manages data access, and performs background management operations.
+
+   - A container for the Automation Agent. </br>
+     The Automation Agent handles configuring, stopping, and restarting the `mongod` process. The Automation Agent periodically polls the `mongod` to determine status and can deploy changes as needed. 
+1. Writes the Automation configuration as a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) and mounts it to each pod. 
+1. Initiates the Automation Agent, which in turn creates the database configuration and launches the `mongod` process according to your MongoDB resource definition.
+
+<!--
+<img src="" alt="Architecure diagram of the MongoDB Community Kubernetes Operator">
+-->
+
+This architecture maximizes use of the Automation Agent while integrating naturally with Kubernetes to produce a number of benefits.
+
+- MongoDB containers are not tied to the Operator or the Agent, so you can:
+  - Run any MongoDB container.
+  - Use your preferred Linux distribution inside the container.
+  - Update operating system packages on your own schedule.
+- Containers are immutable and have a single responsibility or process, so you can:
+  - Describe and understand each container.
+  - Configure resources independently for easier debugging and triage.
+  - Inspect resources independently, including tailing the logs.
+  - Expose the state of each container.
+- Pods are defined as StatefulSets so they benefit from stable identities.
+- You can upgrade the Operator without restarting either the database or the Automation Agent containers.
+- You can set up a MongoDB Kubernetes cluster offline once you download the Docker containers for the database and Automation Agent.
+
+## Example: MongoDB Version Upgrade
+
+The MongoDB Community Kubernetes Operator uses the Automation Agent to efficiently handle rolling upgrades. The Operator configures the StatefulSet to block Kubernetes from performing native rolling upgrades because the native process can trigger multiple  re-elections in your MongoDB cluster. 
```

Comment:
```suggestion
The MongoDB Community Kubernetes Operator uses the Automation Agent to efficiently handle rolling upgrades. The Operator configures the StatefulSet to block Kubernetes from performing native rolling upgrades because the native process can trigger multiple re-elections in your MongoDB cluster. 
```

Reply:
extra space

Reply:
Good 👀